modules/catalog/tech-notes/applyUpdateFlow.puml

```diff
@@ -0,0 +1,32 @@
+/'
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+'/
+
+@startuml
+title Applying an update
+
+activate UpdateLog
+
+UpdateLog --> CatalogManager ++: Notifies about new updates
+CatalogManager -> CatalogManager : Acquire catalog of version equal to version from update minus 1
+CatalogManager -> CatalogManager : Applies updates to the catalog
+CatalogManager -> CatalogManager : Registers new catalog in a local cache
+CatalogManager -> CatalogManager : Notifies other components about changes in catalog
```

Comment:
seems to need to be mentioned in the documentation that CatalogManager can register listeners and notify these listeners about changes

Reply:
done, mention this is in `Base components description` section (see definition of `CatalogService`)